### ISSUE TITLE
ITreeOptions useTriState no longer working as intended

### ISSUE THREAD

**Issue by rwilmert** [500tech/angular-tree-component#699]
As of version 8.0.2, when ITreeOptions for the tree have useCheckBox set to true and useTriState set to false, selecting a node will select any descended node that is also a leaf.
Intended behavior is for only the node checked to be checked and none of its descendants.
# Typescript
``` Typescript
public nodes: any = [
    {
      name: "root1"
    },
    {
      name: "root2",
      children: [
        { name: "child1" },
        {
          name: "child2", children: [
            { name: "grandchild1" },
            { name: "grandchild2" }
          ]
        }
      ]
    }
  ];
  public options: ITreeOptions = {
    useCheckbox: true,
    useTriState: false
  };
```
# Template
``` html5
<tree-root id="tree3" [nodes]="nodes" [options]="options"></tree-root>
```
# Screenshot
Note: The only checkbox that was clicked was on "root2"
![usetristate false with checkboxes true](https://user-images.githubusercontent.com/46051114/50347936-fcbc9d80-04fc-11e9-91ee-2593d7349368.PNG)

**Comment by adamkleingit** [comment]
Hi,
1. There's indeed a bug in computing the parent selected in non tri-state mode - fixed in 8.0.4
2. Intended behaviour is not what you described. Without `useTriState`, the parent checkbox should still reflect the selection of its children.
3. If you want to achieve completely independeny checkboxes you can use `actionMapping` and override `checkboxClick` to determine what happens when user selects the node (and manage the selection state externally)

**Comment by adamkleingit** [comment]
Sorry, you're right. Expected behaviour in the docs is like what you wrote.
Will be fixed in 8.0.5

**Comment by adamkleingit** [comment]
Published
